extensions/circuit/src/adapters/allocate_frame.rs

```diff
@@ -145,20 +145,26 @@ impl<AB: InteractionBuilder> VmAdapterAir<AB> for AllocateFrameAdapterAirWom {
 
         let timestamp_change = AB::Expr::ONE;
 
-        self.execution_bridge.execute_and_increment_pc::<AB>(
-            ctx.instruction.opcode,
-            [
-                local.dest_reg.into(),
-                local.amount_imm.into(),
-                local.amount_reg.into(),
-                local.amount_imm_or_reg.into(),
-                AB::Expr::ZERO,
-                // TODO: is this always one?
-                AB::Expr::ONE,
-            ],
-            local.from_state,
-            timestamp_change,
-        );
+        self.execution_bridge
+            .execute_and_increment_pc::<AB>(
+                ctx.instruction.opcode,
+                [
+                    local.dest_reg.into(),
+                    local.amount_imm.into(),
+                    local.amount_reg.into(),
+                    local.amount_imm_or_reg.into(),
+                    AB::Expr::ZERO,
+                    // TODO: is this always one?
+                    AB::Expr::ONE,
+                ],
+                local.from_state,
+                timestamp_change.clone(),
+            )
+            .eval(builder, ctx.instruction.is_valid.clone());
```

Comment:
Why there wasn't this `eval` before?

Reply:
But for `allocate_frame` specifically we don't actually need constraints, because the allocation is fully nondeterministically provided by the prover